sources/zafira-web/src/main/java/com/qaprosoft/zafira/web/FileUtilController.java

```diff
@@ -66,7 +71,28 @@ public FileUtilController(
     @PostMapping("api/upload")
     @Override
     public String uploadFile(@RequestHeader("FileType") Type type, @RequestParam("file") MultipartFile file) throws IOException {
-        return uploadService.upload(type, file.getInputStream(), file.getOriginalFilename(), file.getSize());
+        String resourceURL;
+
+        // Performing size (less than 2 MB) and file type (JPG/PNG only) validation for images
+        if (COMMON.equals(type) || USERS.equals(type)) {
+
+            boolean isSizeLessThan2Mb = file.getSize() <= 2_000_000;
+            if (!isSizeLessThan2Mb) {
+                throw new IllegalOperationException(INVALID_FILE_SIZE, "File exceeds permitted size");
+            }
+
+            String contentType = file.getContentType();
+            boolean isFormatImage =  contentType!= null &&
```

Comment:
i don't understand what `isFormatImage` means :) 
Create static array ALLOWED_CONTENT_TYPES, btw do not forget "image/jpg". Simplify if condition to (pseudocode): 
```java
if (ALLOWED_CONTENT_TYPES.contains(file.getContentType()) {}
```

Reply:
MIME standard defines both _JPG_ and _JPEG_ as **image/jpeg**